pkg/selectors/kernel_test.go

```diff
@@ -457,6 +458,275 @@ func TestParseMatchNamespaces(t *testing.T) {
+				0x03, 0x00, 0x00, 0x00, // length == 0x3
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Values[0] == "0:0"
+				0xe8, 0x03, 0x00, 0x00, 0xd0, 0x07, 0x00, 0x00, // Values[1] == "1000:2000"
+				0xfd, 0xff, 0xff, 0x6f, 0xff, 0xff, 0xff, 0xff, // Values[3] == "1879048189:4294967295"
+			},
+		},
+		"cred4": {
+			cred: []v1alpha1.CredentialsSelector{
+				{
+					RUIDs: []v1alpha1.CredIDValues{
+						{
+							Operator: "NotEqual",
+							Values: []string{
+								"0:187904818",
+								"1879048187:1879048187",
+								"1879048189:4294967295",
+							},
+						},
+					},
+					EUIDs: []v1alpha1.CredIDValues{
+						{
+							Operator: "NotEqual",
+							Values: []string{
+								"0:187904818",
+								"1879048187:1879048187",
+								"1879048189:4294967295",
+							},
+						},
+					},
+				},
+			},
+			expected: []byte{
+				0x58, 0x00, 0x00, 0x00, // Length
+				0x05, 0x00, 0x00, 0x00, // flags & FILTER_CRED_RUID & FILTER_CRED_EUID
+				0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // FILTER_CRED_RUID
+				0x04, 0x00, 0x00, 0x00, // Operator NotEqual
+				0x03, 0x00, 0x00, 0x00, // length == 0x3
+				0x00, 0x00, 0x00, 0x00, 0x32, 0x33, 0x33, 0x0b, // Values[0] == "0:187904818"
+				0xfb, 0xff, 0xff, 0x6f, 0xfb, 0xff, 0xff, 0x6f, // Values[1] == "1879048187:1879048187"
+				0xfd, 0xff, 0xff, 0x6f, 0xff, 0xff, 0xff, 0xff, // Values[2] == "1879048189:4294967295"
+				0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // FILTER_CRED_EUID
+				0x04, 0x00, 0x00, 0x00, // Operator NotEqual
+				0x03, 0x00, 0x00, 0x00, // length == 0x3
+				0x00, 0x00, 0x00, 0x00, 0x32, 0x33, 0x33, 0x0b, // Values[0] == "0:187904818"
+				0xfb, 0xff, 0xff, 0x6f, 0xfb, 0xff, 0xff, 0x6f, // Values[1] == "1879048187:1879048187"
+				0xfd, 0xff, 0xff, 0x6f, 0xff, 0xff, 0xff, 0xff, // Values[2] == "1879048189:4294967295"
+			},
+		},
+		"cred5": {
+			cred: []v1alpha1.CredentialsSelector{
+				{
+					RUIDs: []v1alpha1.CredIDValues{
+						{
+							Operator: "Equal", Values: []string{"1:0"},
+						},
+					},
+				},
+			},
+			err: errors.New("invalid range"),
+		},
+		"cred6": {
+			cred: []v1alpha1.CredentialsSelector{
+				{
+					RUIDs: []v1alpha1.CredIDValues{
+						{
+							Operator: "Equal", Values: []string{"1:4294967296"},
+						},
+					},
+				},
+			},
+			err: errors.New("invalid range"),
+		},
+		"cred7": {
+			cred: []v1alpha1.CredentialsSelector{
+				{
+					RUIDs: []v1alpha1.CredIDValues{
+						{
+							Operator: "Equal", Values: []string{"-1:0"},
+						},
+					},
+				},
+			},
+			err: errors.New("invalid range"),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			k := &KernelSelectorState{data: KernelSelectorData{off: 0}}
+			d := &k.data
+			err := ParseMatchCurrentCred(k, test.cred)
+			if test.err != nil {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.True(t, bytes.Equal(test.expected, d.e[0:d.off]))
+			}
+
+		})
+	}
+
+	/* Extra tests */
```

Comment:
Can you please use a similar structure as above? That is, something like:

```
tests := map[string]struct {
		cred     []v1alpha1.CredentialsSelector
		expected []byte
		err      error
```

Reply:
Will try otherwise I separate them in different test, since they are a bit special...